platform/mds/mds-test-bundle/src/test/java/org/motechproject/mds/test/osgi/MdsDdeBundleIT.java

```diff
@@ -470,6 +488,51 @@ public void testSetOfEnumsAndSetOfStrings() {
         assertTrue(CollectionUtils.contains(retrieved.getRecipients().iterator(), "Christina"));
     }
 
+    @Test 
+    public void testCascadeDelete2() {
+        EntityX x1 = new EntityX();
+        EntityX x2 = new EntityX();
+        
+        EntityZ z1 = new EntityZ();
+        EntityZ z2 = new EntityZ();
+        
+        EntityY y1 = new EntityY();
+        EntityY y2 = new EntityY();
+        
+        x1.getY().add(y1);
+        x1.getY().add(y2);
+        y2.getX().add(x1);
+        y2.getZ().add(z2);
+        y1.getX().add(x1);
+        y1.getZ().add(z1);
+        
+        entityXDataService.create(x1);
```

Comment:
Creating x1 automatically creates all its relationships so further creates should be removed.